plugins/inputs/statsd/statsd.go

```diff
@@ -807,7 +823,17 @@ func (s *Statsd) handler(conn *net.TCPConn, id string) {
 		s.forget(id)
 		s.CurrentConnections.Incr(-1)
 	}()
-
+	addr := conn.RemoteAddr()
+	a, err := url.Parse(addr.String())
+	var host string
+	if err != nil {
+		// this should never happen because the conn handler should give us parsable addresses,
+		// but if it does we will know
+		host = "badhost"
+		log.Printf("E! failed to parse %s\n", addr)
```

Comment:
Either close the connection or drop the packet (tcp vs udp), or just silently don't set a remote host.

Reply:
In the udp case, there isn't a possibility of it not parsing, because the address it gives us is *net.UDPAddr, so we can pull out the IP directly.
I'll modify the TCP case. though